examples/task_controller_client/section_control_implement_sim.cpp

```diff
@@ -139,10 +139,10 @@ bool SectionControlImplementSimulator::create_ddop(std::shared_ptr<isobus::Devic
 	// Add child linkages to device elements if all objects were added OK
 	if (retVal)
 	{
-		auto sprayer = reinterpret_cast<isobus::task_controller_object::DeviceElementObject *>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::MainDeviceElement)));
-		auto connector = reinterpret_cast<isobus::task_controller_object::DeviceElementObject *>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::Connector)));
-		auto boom = reinterpret_cast<isobus::task_controller_object::DeviceElementObject *>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::SprayBoom)));
-		auto product = reinterpret_cast<isobus::task_controller_object::DeviceElementObject *>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::LiquidProduct)));
+		auto sprayer = std::static_pointer_cast<isobus::task_controller_object::DeviceElementObject>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::MainDeviceElement)).lock());
+		auto connector = std::static_pointer_cast<isobus::task_controller_object::DeviceElementObject>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::Connector)).lock());
+		auto boom = std::static_pointer_cast<isobus::task_controller_object::DeviceElementObject>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::SprayBoom)).lock());
+		auto product = std::static_pointer_cast<isobus::task_controller_object::DeviceElementObject>(poolToPopulate->get_object_by_id(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::LiquidProduct)).lock());
```

Comment:
Actually, it might be better to return a `shared_ptr` instead for the `get_object_by_id` maybe? Since you'd have to check if `.lock()`is actually returning something everytime otherwise. And as far as I can tell with a task controller object, you don't have to worry about it being in a 'valid' state since it's standalone I imagine?

Reply:
Well, you'll still have to check the shared_ptr isn't nullptr, as it would return an empty shared_ptr in the case that an object is not found. But, it would make for less use of `lock` when it will almost always be valid

Reply:
Ah I see, that's right. If only we were using Rust where null/nullptr doesn't exist haha... 😛

Reply:
Do you still think I should change it, just to avoid `lock`? or leave it?

Reply:
Uhm, I don't see any cons of changing from weak to shared. Avoiding `lock` here seems to me like less confusion can occur, so yea I think changing it is still worth it

Reply:
OK! I will make that change, thanks for your feedback 😁